specification/src/lib.rs

```diff
@@ -159,6 +160,19 @@ pub trait EbmlTag<T: Clone> {
     /// This function *must* return `None` if the associated data type of `self` is not [`TagDataType::Master`].  Implementors can reference [webm-iterable](https://crates.io/crates/webm_iterable) for an example.
     ///
     fn as_master(&self) -> Option<&Master<T>>;
+
+    ///
+    /// Tests if [id] is a child of self
+    ///
+    /// Default is [true]
+    ///
+    /// Unknown or unexpected elements are children. Global elements are children.
+    ///
+    /// Used to determine if Unknown size blocks should end. If [is_child] returns false, the Unknown size block ends.
+    ///
+    /// [`Spec`](https://github.com/ietf-wg-cellar/ebml-specification/blob/master/specification.markdown#unknown-data-size) does not explicitly state what to do in case of an unexpected element, so we assume it will be handled like a child element since it is not one of the end conditions.
+    ///
+    fn is_child(&self, id: u64) -> bool;
```

Comment:
I think it may be simpler to implement a `get_parent_id` method here.  Rather than building out an exhaustive list of "NOT_CHILDREN" elements in the spec derivation, you would only need to return the parent id described in the new "parent" attribute.

The spec says that the end of an Unknown-sized element is determined by the first (paraphrasing):
- Parent of the unknown-sized element
- Sibling of the unknown-sized element
- Valid "root" element
- End of parent element with known size
- End of file

The first three conditions can all be easily checked in the tag iterator if we can determine any element's parent.  We can call get_parent_id recursively on the unknown-sized element to determine whether the new element is in its parent chain; we can call get_parent_id on the unknown-sized element and on the new element to determine whether they are siblings; if get_parent_id of the new element is `None`, that would indicate a root element.  (Unless the tag is a RawTag variant, in which case I think it shouldn't end the unknown-sized tag)

IMO this approach would simplify the macro code and make it more readable.  It would also keep all of the spec implementation logic closer together inside the tag_iterator rather than making a parent/sibling/root/child determination in the macro code.

```suggestion
    fn get_parent_id(&self) -> Option<u64>;
```

Reply:
the problem arises with malformed files, especially during streams, the spec is very clear that the unknown element should not end the stream as it specifies exactly the conditions for the end. This is due to the extensibility of the format, we are maybe not aware of an extension.

Reply:
You're right - that case applies to my comment about the RawTag variant not ending a stream.  I'll put together some code to show exactly what I meant in this comment, but probably won't be able get to it until late tomorrow or Friday

Reply:
Feel free to change my code, i'm just making my work available since yours spared me some time. also i took the liberty to change the generated if forests to match arms in the dreive crate.